blosc/blosc2.c

```diff
@@ -1373,18 +1373,19 @@ static void flags_to_filters(const uint8_t flags, uint8_t* filters) {
 
 
 static int initialize_context_compression(
-  blosc2_context* context, int32_t sourcesize, const void* src, void* dest,
+  blosc2_context* context, const void* src, size_t srcsize, void* dest,
```

Comment:
You are using `size_t` here for `srcsize`, whereas `destsize` uses `int32_t`.  I am very much inclined to use explicit type sizes, so I'd suggest to converge both to `int32_t`.

Reply:
For some reason `get_coffsets` uses `int64_t` in `cbytes` and `off_cbytes`. Can these be changed to `int32_t`? It seems like size variables can be `int32_t` but position variables need to be `int64_t`..

Reply:
Yes, `get_coffsets` are designed to get offsets within frames, which are meant as 63-bit containers rather than 31.  I could have stored 31-bit chun lengths as the frame index, and then compute absolute offsets during reading, but I thought that storing 63-bit indexes would be easier (and compression would get rid of the additional storage).

Reply:
The variables though are not representing offsets, it seems they are representing number of compressed bytes, so they can be `int32_t`?

Reply:
Otherwise I am not sure why `cbytes` is allowed to be `int64_t` but yet we can only ever pass in `int32_t` into `blosc2_compress_ctx`. 

https://github.com/Blosc/c-blosc2/blob/f90c05489f5c5d4335f2a5e3338f57c9e0c13e4a/blosc/frame.c#L987

Reply:
Sorry, I know this can be confusing for the newcomer.  The `cbytes` and `off_cbytes` from `get_coffsets` params refers to the length of the frame and offsets to each chunk inside the frame, so they are `int64_t` indeed.  For the sake of readibility, perhaps `cbytes` here should be renamed to something like `frame_len` or similar.  Hope this helps.

Reply:
In the code in frame.c, We get the size of the `coffset_cbytes` which is `int64_t`. Then should we cast it to `int32_t` when calling `blosc2_decompress_ctx`?

```
  uint8_t* coffsets = get_coffsets(frame, header_len, cbytes, &coffsets_cbytes);
  ...
  int32_t off_nbytes = blosc2_decompress_ctx(dctx, coffsets, (int32_t)coffsets_cbytes, offsets, nchunks * 8);
```
Or should `srcsize` and `destsize` be `int64_t`?

Reply:
After close inspection, `coffset_cbytes` is the amount of compressed bytes of the offsets chunk.  This number can never be larger than 2**31, so this is clearly an `int32_t`.  Sorry that I did not detected this when I merged your previous PR.  Could you fix it please?